provider/resource_keycloak_required_action.go

```diff
@@ -0,0 +1,129 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func resourceKeycloakRequiredAction() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceKeycloakRequiredActionsCreate,
+		Read:   resourceKeycloakRequiredActionsRead,
+		Delete: resourceKeycloakRequiredActionsDelete,
+		Update: resourceKeycloakRequiredActionsUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```

Comment:
You are right, It does not work,
Do you have already have an example/other location, where you set the realm in the importer function?

Reply:
I made the import function. But I am not 100% sure that I did everything correct.
Please make a thorough review.
I do not think I fully understand with it is supposed to do. Terraform does not really support importing if I am not mistaken. Yet here we are creating a config and checking if an import works?

Reply:
Yeah importing with TF is a little weird to implement.  The way I understand it, Terraform doesn't actually look at your HCL when importing (except I believe it still validates things like required attributes).  Instead, Terraform uses the information provided to it in the last argument to `terraform import` to set all of the resource data required for the read function to operate correctly.  This means that you have to give Terraform everything it needs to know in that last argument, which is usually done by specifying a string delimited by `/` characters.

A good example of this is the [`keycloak_ldap_user_federation` resource](https://github.com/mrparkers/terraform-provider-keycloak/blob/master/provider/resource_keycloak_ldap_user_federation.go#L364-L384).  This resource is tied to a realm, and it has its own unique ID, meaning that you need to specify a string like `{{realm_id}}/{{ldap_user_federation_id}}`.  These attributes are saved in data, and the provider performs the `read` function, which only requires these two attributes to fetch the remote resource from Keycloak.  Optionally, you can also specify a third argument for the `bind_credential` attribute, because the Keycloak API does not ever return this value, so this is the only way to save it into state when importing.

In your case, your read function only requires `realm_name` (or `realm_id` if you rename this attribute) and `alias` to function properly.  This means that your import function needs perform `d.Set("alias", ...)` and `d.Set("realm_id", ...)`.

Does this make sense?

Reply:
@mrparkers I still do not understand this completely, For example where is it decided that the data.Id() exists in 2 parts and can be split at '/' 

For example the keycloak_ldap_user_federation resource, the id is everywhere just {{ldap_user_federation_id}} 
But suddenly at import the id is {{realm_id}}/{{ldap_user_federation_id}},
How does this happen? Where is the set?
I need to understand this for authenticationSubFlows and authenticationExecutions.
I have an import test failing, And I think it might be because it is missing the parentFlowAlias.
So how do I add parentFlowAlias to the import id....... {{realmId}}/ {{parentFlowAlias}}/{{id}}

>  This means that you have to give Terraform everything it needs to know in that last argument, which is usually done by specifying a string delimited by / characters.

Where do you specify this?